src/figuresClasses.ts

```diff
@@ -1,19 +1,74 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: Function,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const data = [a, b, c].sort((prev, next) => next - prev);
 
-}
+    if (a <= 0 || b <= 0 || c <= 0 || data[0] >= data[1] + data[2]) {
+      throw new Error('your error message');
+    }
+  }
+
+  shape: Shape = 'triangle';
+
+  getArea(): number {
+    const p: number = (this.a + this.b + this.c) * 0.5;
 
-export class Circle {
+    const s:number = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
 
+    return +s.toFixed(2);
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  constructor(
+    public color: Color,
+    public radius : number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('your error message');
```

Comment:
```suggestion
      throw new Error('radius must be greater than 0');
```
Потрібно вказувати, в чому саме помилка